src/background/Wallet/WalletRecord.ts

```diff
@@ -727,6 +727,19 @@ export class WalletRecordModel {
       .map(([origin, permission]) => ({ origin, permission }));
   }
 
+  static removeChainFromPermissions(
+    record: WalletRecord,
+    { chain }: { chain: Chain }
+  ) {
+    return produce(record, (draft) => {
+      draft.permissions = Object.fromEntries(
+        Object.entries(record.permissions)
+          .filter(([, permission]) => permission.chain === chain.toString())
```

Comment:
As I understand, this method will remove all permissions with other chains completely.
Cause we need to keep all permissions for other chains + change filtered permissions and remove chain for them

Reply:
thanks for catching this, what a shame %)